prime-router/src/main/kotlin/azure/DownloadFunction.kt

```diff
@@ -88,63 +89,62 @@ class DownloadFunction(private val workflowEngine: WorkflowEngine = WorkflowEngi
+        return reportFiles.sortedByDescending {
+            it.createdAt
         }.map {
-            val svc = WorkflowEngine().settings.findReceiver(it.task.receiverName)
+            val svc = WorkflowEngine().settings.findReceiver(it.receivingOrg)
             val orgDesc = authClaims.organization.description
             val receiver = if (svc !== null && svc.description.isNotBlank()) svc.description else orgDesc
             TestResult(
-                it.task.createdAt.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")),
+                it.createdAt.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")),
                 receiver,
-                DAYS_TO_SHOW - it.task.createdAt.until(OffsetDateTime.now(), ChronoUnit.DAYS),
-                it.task.itemCount,
+                DAYS_TO_SHOW - it.createdAt.until(OffsetDateTime.now(), ChronoUnit.DAYS),
+                it.itemCount,
                 0,
-                it.task.reportId.toString(),
-                it.task.bodyFormat
+                it.reportId.toString(),
+                it.bodyFormat
             )
         }
     }
 
     private fun generateTodaysTestResults(
-        headers: List<WorkflowEngine.Header>,
+        reportFiles: List<ReportFile>,
         authClaims: AuthClaims
     ): List<TestResult> {
-        var filtered = headers.filter { filter(it) }
+        var filtered = reportFiles.filter { filter(it) }
         return generateTestResults(filtered, authClaims)
     }
 
-    private fun filter(it: WorkflowEngine.Header): Boolean {
+    private fun filter(reportFile: ReportFile): Boolean {
         val now = OffsetDateTime.now()
-        return it.task.createdAt.year == now.year &&
-            it.task.createdAt.monthValue == now.monthValue &&
-            it.task.createdAt.dayOfMonth == now.dayOfMonth
+        return reportFile.createdAt.year == now.year &&
+            reportFile.createdAt.monthValue == now.monthValue &&
+            reportFile.createdAt.dayOfMonth == now.dayOfMonth
     }
 
     private fun generatePreviousTestResults(
-        headers: List<WorkflowEngine.Header>,
+        reportFiles: List<ReportFile>,
         authClaims: AuthClaims
     ): List<TestResult> {
-        var filtered = headers.filterNot { filter(it) }
+        var filtered = reportFiles.filterNot { filter(it) }
         return generateTestResults(filtered, authClaims)
     }
 
     private fun responsePage(
         request: HttpRequestMessage<String?>,
-        authClaims: AuthClaims,
-        context: ExecutionContext
+        authClaims: AuthClaims
     ): HttpResponseMessage {
         val htmlTemplate: String = Files.readString(Path.of(DOWNLOAD_PAGE))
-        val headers = workflowEngine.fetchDownloadableHeaders(
+        val reportFiles = workflowEngine.fetchDownloadableReportFiles(
             OffsetDateTime.now().minusDays(DAYS_TO_SHOW), authClaims.organization.name
```

Comment:
@mry-usds is right - you don't want to pull in the whole Header for the list of files, because it now includes the onerous ItemLineage stuff (which could possibly be lazy loaded, but is not at this time).  The ReportFile table by design contains everything you need to know about a report - so just pull in that.